src/main/asciidoc/transactions.adoc

```diff
@@ -33,82 +41,257 @@ static class Config extends AbstractCouchbaseConfiguration {
 
-.Transaction Conversion on Write
+Note that a CouchbaseTransactionalOperator is available as a bean.
+If creating one explicitly, it must be created with the following (and not TransactionalOperator.create(manager)).
+.TransactionalOperator Construction
 ====
 [source,java]
 ----
-@Autowired
-MappingCouchbaseConverter mappingCouchbaseConverter;
+public static CouchbaseTransactionalOperator create(CouchbaseCallbackTransactionManager manager) 
+====
 
-public void doSomething() {
-  transactions.run(ctx -> {
+.Transaction Access Using TransactionalOperator.execute()
+====
+[source,java]
+----
+@Autowired TransactionalOperator txOperator;
 
-   Airline airline = new Airline("demo-airline", "at");
-   CouchbaseDocument target = new CouchbaseDocument();
-   mappingCouchbaseConverter.write(airline, target);
+    Person person = rxCBTmpl.insertById(Person.class).inCollection(cName).one(WalterWhite).block();
+    Flux<Person> result = txOperator.execute((ctx) -> rxCBTmpl.findById(Person.class).one(person.id())
+        .flatMap(p -> rxCBTmpl.replaceById(Person.class).one(p.withFirstName("Walt"))));
+    result.blockLast();
+----
+====
 
-   ctx.insert(couchbaseClientFactory.getDefaultCollection(), target.getId(), target.getContent());
+.Transaction Access Using transactional()
+====
+[source,java]
+----
+@Autowired TransactionalOperator txOperator;
 
-   ctx.commit();
-  });
-}
+    Person person = rxCBTmpl.insertById(Person.class).inCollection(cName).one(WalterWhite).block();
+    Mono<Person> result = rxCBTmpl.findById(Person.class).one(person.id())
+        .flatMap(p -> rxCBTmpl.replaceById(Person.class).one(p.withFirstName("Walt")))
+        .as(txOperator::transactional);
+    result.block();
 ----
 ====
+.Transaction Access  simulating @Transactional
+====
+[source,java]
+----
+@Service
+@Component
+class PersonService {
+
+    final ReactiveCouchbaseOperations personOperationsRx;
+    final TransactionalOperator transactionalOperator;
+
+    public PersonService(ReactiveCouchbaseOperations opsRx, TransactionalOperator transactionalOperator) {
+	this.personOperationsRx = opsRx;
+        this.transactionalOperator = transactionalOperator;
+    }
 
-The same approach can be used on read:
+    // this method does not have the @Transactional annotation, 
+    // but the as(trasnactionalOperator::transactional)
+    // makes it behave as though it does.
+    public Mono<Person> declarativeFindReplacePersonReactive(Person person) {
+        return personOperationsRx.findById(Person.class).one(person.id())
+                .flatMap(p -> personOperationsRx.replaceById(Person.class).one(p.withFirstName(person.getFirstname())))
+                .as(transactionalOperator::transactional);
+    }
+----
+====
 
-.Transaction Conversion on Read
+== Transactions with Only the SDK
```

Comment:
Oh wow, this sections is very barebones :)

I feel this should be our best seller.  We made the regular transactions API support all Spring operations too - that's really cool!  I think this should be the main way that users interact with transactions in Spring.  We should have this front and centre, and mention @Transactional second - IMO.

Can this section be fleshed out a lot more?  I'd like to see a good description section, and examples including:

- Doing correct error handling of TransactionCommitAmbiguousException and TransactionFailedException (see the docs)
- Mixing the normal transactions API with Spring repo and template methods.

Reply:
Can we not call it "Only the SDK", it just sounds a bit negative :)
Maybe "Using the Transactions SDK Directly" ?  Or something, I dunno, naming things is hard.

Reply:
I still feel like we could talk this section up a bit, it's incredibly barebones.  But I guess we just disagree on this one.

Reply:
I looked at what Maria did with the main transactions documentation.   I want her to give this a going-over when she gets a chance. (also the whole spring-data-couchbase documentation - it's all very ... minimal).

Reply:
Yeah that's a great idea, Maria will likely have some good suggestions.

Reply:
I stuck in some extra text to brighten it up.